[FX] Groups rework

tobiasdiez opened the pull request:
<!-- describe the changes you have made here: what, why, ... -->
Continuation of https://github.com/JabRef/jabref/pull/1438 (for some reason github didn't let me reopen it, thus a new PR).

The core functionality now works, that is:
- show groups tree
- filter entries when a group is selected
- show number of entries (well, not really "core" but works nonetheless)

I tested it with a big db and the performance was quite good. 
Of course, many features of the original dialog are still missing. But I think it is better to merge/rework the group pane peu à peu instead of a big un-reviewable PR. Also feedback about the design is welcome (this is just a proposal). [Builds are available](http://builds.jabref.org/groupsRework/) since I learned how to push to master 😄 . 

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

AEgit commented:
Coming from: https://github.com/JabRef/jabref/issues/2404

I've just installed JabRef_windows-x64_3_8_1-dev--snapshot--2016-12-20--groupsRework--d78d7ae: The groups interface appears to be empty/just gray. Furthermore "About JabRef" does not seem to work (no window appears).

I have Java 1.8.0_111: Do I need a specific Java version for JavaFX to run?

tobiasdiez commented:
@AEgit Thanks for trying out the build. I just updated to Java 1.8.111 and had similar problems with the new groups. A fixed version should be available in a few minutes under http://builds.jabref.org/groupsRework/. However, the "About" dialog worked fine for my. Could you check under "Help > Error console" whether some error messages are reported (but this dialog is also implemented in JavaFX, so if you have general problems with it than probably this also doesn't work and you would need to run [JabRef from the console](http://help.jabref.org/en/CommandLine) to see error messages)

AEgit commented:
Ok, tried the new version, but I still have the same problems. I ran it from command line. When I try to display "About JabRef" the following error message appears:

```
File: aboutdialog.fxml not found, attempting with camel case
File: aboutdialog.css not found, attempting with camel case
16:32:20.820 [JavaFX Application Thread] ERROR net.sf.jabref.FallbackExceptionHandler - Uncaught exception Occurred in Thread[JavaFX Application Thread,5,main]
java.lang.NullPointerException: null
        at net.sf.jabref.gui.AbstractView.getView(AbstractView.java:25) ~[JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.help.AboutDialogView.show(AboutDialogView.java:15) ~[JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.help.AboutAction.lambda$actionPerformed$0(AboutAction.java:22) ~[JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295) ~[jfxrt.jar:?]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_111]
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294) ~[jfxrt.jar:?]
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95) ~[jfxrt.jar:?]
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method) ~[jfxrt.jar:?]
        at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191) ~[jfxrt.jar:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
```

When trying to display the "Groups panel" I get the following error message:

```
16:36:56.253 [AWT-EventQueue-0] ERROR net.sf.jabref.gui.SidePaneManager - Following side pane could not be found: Internetrecherche
java.lang.ClassNotFoundException: Internetrecherche
        at java.net.URLClassLoader.findClass(Unknown Source) ~[?:1.8.0_111]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_111]
        at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source) ~[?:1.8.0_111]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_111]
        at java.lang.Class.forName0(Native Method) ~[?:1.8.0_111]
        at java.lang.Class.forName(Unknown Source) ~[?:1.8.0_111]
        at net.sf.jabref.gui.SidePaneManager.getPreferredPositions(SidePaneManager.java:167) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.access$000(SidePaneManager.java:25) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager$PreferredIndexSort.<init>(SidePaneManager.java:213) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.show(SidePaneManager.java:124) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.show(SidePaneManager.java:96) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.toggleThreeWay(SidePaneManager.java:87) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneComponent$ToggleAction.actionPerformed(SidePaneComponent.java:142) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at javax.swing.AbstractButton.fireActionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.JToggleButton$ToggleButtonModel.setPressed(Unknown Source) [?:1.8.0_111]
        at javax.swing.AbstractButton.doClick(Unknown Source) [?:1.8.0_111]
        at javax.swing.AbstractButton.doClick(Unknown Source) [?:1.8.0_111]
        at javax.swing.plaf.basic.BasicMenuItemUI$Actions.actionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.SwingUtilities.notifyAction(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyBinding(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processKeyBinding(Unknown Source) [?:1.8.0_111]
        at javax.swing.KeyboardManager.fireBinding(Unknown Source) [?:1.8.0_111]
        at javax.swing.KeyboardManager.fireKeyboardAction(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyBindingsForAllComponents(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyBindings(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.processEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Container.processEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.KeyboardFocusManager.redispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Window.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue.access$500(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_111]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_111]
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_111]
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_111]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_111]
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.run(Unknown Source) [?:1.8.0_111]
16:36:56.258 [AWT-EventQueue-0] ERROR net.sf.jabref.gui.SidePaneManager - Following side pane could not be found: groups
java.lang.ClassNotFoundException: groups
        at java.net.URLClassLoader.findClass(Unknown Source) ~[?:1.8.0_111]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_111]
        at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source) ~[?:1.8.0_111]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_111]
        at java.lang.Class.forName0(Native Method) ~[?:1.8.0_111]
        at java.lang.Class.forName(Unknown Source) ~[?:1.8.0_111]
        at net.sf.jabref.gui.SidePaneManager.getPreferredPositions(SidePaneManager.java:167) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.access$000(SidePaneManager.java:25) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager$PreferredIndexSort.<init>(SidePaneManager.java:213) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.show(SidePaneManager.java:124) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.show(SidePaneManager.java:96) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneManager.toggleThreeWay(SidePaneManager.java:87) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at net.sf.jabref.gui.SidePaneComponent$ToggleAction.actionPerformed(SidePaneComponent.java:142) [JabRef-3.8.1-dev--snapshot--2016-12-22--groupsRework--69951e8.jar:?]
        at javax.swing.AbstractButton.fireActionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.JToggleButton$ToggleButtonModel.setPressed(Unknown Source) [?:1.8.0_111]
        at javax.swing.AbstractButton.doClick(Unknown Source) [?:1.8.0_111]
        at javax.swing.AbstractButton.doClick(Unknown Source) [?:1.8.0_111]
        at javax.swing.plaf.basic.BasicMenuItemUI$Actions.actionPerformed(Unknown Source) [?:1.8.0_111]
        at javax.swing.SwingUtilities.notifyAction(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyBinding(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(Unknown Source) [?:1.8.0_111]
        at javax.swing.JMenuBar.processKeyBinding(Unknown Source) [?:1.8.0_111]
        at javax.swing.KeyboardManager.fireBinding(Unknown Source) [?:1.8.0_111]
        at javax.swing.KeyboardManager.fireKeyboardAction(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyBindingsForAllComponents(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyBindings(Unknown Source) [?:1.8.0_111]
        at javax.swing.JComponent.processKeyEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.processEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Container.processEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.KeyboardFocusManager.redispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(Unknown Source) [?:1.8.0_111]
        at java.awt.DefaultKeyboardFocusManager.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Container.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Window.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.Component.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue.access$500(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_111]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_111]
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_111]
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue$4.run(Unknown Source) [?:1.8.0_111]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_111]
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_111]
        at java.awt.EventQueue.dispatchEvent(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_111]
        at java.awt.EventDispatchThread.run(Unknown Source) [?:1.8.0_111]
```
